src/api/article/controller.js

```diff
@@ -61,67 +64,173 @@ export const getOne = ({ params: { slug }, user }, res, next) =>
+                    'authorEmail',
+                    'brand',
+                    'brandSlug',
+                    'collectionSlug',
+                    'locales',
+                  ]);
+                  // TODO(uladbohdan): to refactor that, i.e. with pickBy / omitBy.
+                  if (newBrand) {
+                    defaultFields.brand = newBrand._id;
+                  }
+                  if (newCollection) {
+                    defaultFields.collectionId = newCollection._id;
+                  }
+                  if (newAuthor) {
+                    defaultFields.author = newAuthor._id;
+                  }
+
+                  return defaultFields;
+                })
+                .then(updateFields =>
+                  Article.findOneAndUpdate({ _id: articleId }, updateFields, { new: true })
+                    .then(checkIsFound)
+                    .then(async article => {
+                      // Here Article is properly updated. Proceeding with locales.
+                      let articleOldLocales = [];
+                      if (body.locales) {
+                        articleOldLocales = article.locales.slice(); // Copying.
+                        article.locales.splice(0, article.locales.length); // Clearing with no reassignment.
+                        await Promise.all(
+                          Object.keys(body.locales).map(updLoc =>
+                            LocalizedArticle.findOneAndUpdate(
+                              {
+                                locale: updLoc,
+                                articleId,
+                              },
+                              body.locales[updLoc],
+                              { new: true }
+                            )
+                              .then(async updatedLocalization => {
+                                if (!updatedLocalization) {
+                                  // Was not found + updated. Must be created. I assume.
+                                  const newLocalization = await LocalizedArticle({
+                                    ...body.locales[updLoc],
+                                    articleId,
+                                  }).save();
+                                  return newLocalization._id;
+                                }
+                                return updatedLocalization._id;
+                              })
+                              .then(localeId => article.locales.push(localeId))
+                              .catch(next)
+                          )
+                        );
+                      }
+                      return { article, articleOldLocales };
+                    })
+                    .then(({ article, articleOldLocales }) =>
+                      article.save().then(async () => {
+                        // Here I want to remove unnecessary locales.
+                        // All locales which are in articleOldLocales and are not in article.locales
```

Comment:
```js
const toRemove = articleOldLocales.filter(l => !article.locales.includes(s))
```

Reply:
How difference is worse?

Reply:
for me `filter` is more clear and understandable